src/DependencyInjection/Configuration.php

```diff
@@ -629,10 +629,10 @@ private function addSharedPluginNodes(ArrayNodeDefinition $pluginNode, $disableA
             ->canBeEnabled()
             ->addDefaultsIfNotSet()
             ->children()
-                ->scalarNode('name')->end()
-                ->scalarNode('key')->defaultNull()->end()
-                ->integerNode('tokens')->defaultValue(1)->end()
-                ->floatNode('max_time')->defaultNull()->end()
+                ->scalarNode('name')->isRequired()->info('Rate limiter configuration name from rate_limiter.yaml')->end()
```

Comment:
this will tricky when a user configures their own service for a rate limiter factory and does not call it `limiter.{name}`. 

in other places, we ask for a full service name for things. i think i would prefer to do the same here:
```suggestion
                ->scalarNode('name')->isRequired()->info('Rate limiter factory service name, e.g. limiter.http_client for a rate limiter called http_client')->end()
```

wdyt? if we change it here, we also need to adjust the extension class.

and now that i understand what it is, it should be called more specific `limiter_name`. or while we do change it to be the service,  `limiter_service`.

Reply:
@dbu Symfony automatic create factory in container, I just have this config
![image](https://github.com/php-http/HttplugBundle/assets/646054/151714ed-65e5-46d5-a76f-81d741cd4cc7)

I think using just "name" more understandable than whole name in container, like "limiter.http_client"

Reply:
yes, when one configures in framework bundle, that creates the services with this naming pattern.

the thing is that a user might configure their own service rather than use the framework configuration, e.g. when they have a custom implementation of the rate limiter. and for that i think its more consistent with the rest of the bundle to have the full service name.